features/src/autogluon/features/generators/datetime.py

```diff
@@ -39,42 +39,54 @@ def _transform(self, X: DataFrame) -> DataFrame:
 
     @staticmethod
     def get_default_infer_features_in_args() -> dict:
-        return dict(required_raw_special_pairs=[
-            (R_DATETIME, None),
-            (None, [S_DATETIME_AS_OBJECT])
-        ])
+        return dict()
 
     def _compute_fillna_map(self, X: DataFrame):
         fillna_map = dict()
         for datetime_feature in self.features_in:
-            datetime_series = pd.to_datetime(X[datetime_feature], errors='coerce')
+            if type(X[datetime_feature].iloc[0]) == pd.Timestamp:
+                datetime_series = pd.to_datetime(X[datetime_feature], errors='coerce')
 
-            # Best guess is currently to fill by the mean.
-            fillna_datetime = datetime_series.mean()
-            fillna_map[datetime_feature] = fillna_datetime
+                # Best guess is currently to fill by the mean.
+                fillna_datetime = datetime_series.mean()
+                fillna_map[datetime_feature] = fillna_datetime
         return fillna_map
 
     # TODO: Improve handling of missing datetimes
     def _generate_features_datetime(self, X: DataFrame) -> DataFrame:
-        X_datetime = DataFrame(index=X.index)
+        X_datetime = X.copy()
+        country = locale.getlocale()[0].split("_")[1] #　Obtain the country used to determine if it is a holiday or not.
```

Comment:
If this gets the locale of the machine, this is dangerous. If someone is training on the cloud, the locale could be different than what they intend. I'd prefer that we don't use this kind of information as it means reproducing results will be impossible between two different regions.

Reply:
In deciding which country should be used, I used environment variables.